nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManagerComponent.java

```diff
@@ -262,6 +268,9 @@ public BinaryManagerStatus garbageCollectBinaries(boolean delete) {
             globalStatus.sizeBinariesGC += status.sizeBinariesGC;
         }
         globalStatus.gcDuration = System.currentTimeMillis() - start;
+        // go back to default transaction timeout
+        TransactionHelper.commitOrRollbackTransaction();
+        TransactionHelper.startTransaction();
```

Comment:
Before starting a new transaction, shouldn't we check that we had a transaction when entering the `#garbageCollectBinaries` method?

Seems weird to me to start a new transaction if there was no transaction before.

Reply:
What about a 
```java
TransactionHelper.runInNewTransaction(Supplier<R> supplier, long txTimeout)
```
?

Reply:
> It seems weird to me to start a new transaction if there was no transaction before.

That's the purpose of the fix: opening a transaction with a long timeout when there is none, which is the case when calling the binary GC from the JSF UI Admin Center, to avoid subsequent calls to the DB session being run in dedicated transactions with a short timeout, hardly customizable.

Then, the problem might be when exiting the method where we start new transaction whereas the method wasn't called inside a transaction.
Let's wait for @efge's opinion.

About the new API in `TransactionHelper`, not sure what's the benfit of it since this shoudn't be a frequent use case. IMHO, either one increases the global TX, either we make sure in our code that any long-running work is done outside of a transaction, or commited by batch with a bulk action.

Reply:
Yes ok let's be careful and check status before. But please don't use `runInNewTransaction` as the idea of that one is to *keep* the existing transaction if it exists so it potentially has to start a new thread. Anything using `runWithoutTransactionInternal` is bad, I should never have added this API

Reply:
Refactored to handle the case where there was no transaction before.

Done in a helper method local to the `DocumentBlobManagerComponent` class. We already have too many methods in `TransactionHelper`, which reflects the fact that we are not handling our transactional contexts well enough at the moment.